server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -315,6 +315,29 @@ public DirectoryReader getDirectoryReader() {
         return (DirectoryReader) reader;
     }
 
+    /**
+     * Returns the total hit count from index reader leaves and a query using the count function.
+     *
+     * @param reader an index reader.
+     * @param query the query to compute hit count.
+     * @return the total hit count of a given query.
+     * @throws IOException if there is a low-level I/O error
+     */
+    public int getTotalHitCount(IndexReader reader, Query query) throws IOException {
+        int hitCount = -1;
+        Weight weight = createWeight(query, ScoreMode.COMPLETE_NO_SCORES, 1f);
+        for (LeafReaderContext leaf : reader.leaves()) {
+            int count = weight.count(leaf);
+            if (count != -1) {
```

Comment:
let's return -1 if `count` returns -1 on any of the segments, to mean that the count is unknown?